relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -1,20 +1,473 @@
 use std::collections::BTreeMap;
 
-use relay_common::{SpanStatus, UnixTimestamp};
+use relay_common::{MetricUnit, SpanStatus, UnixTimestamp};
 use relay_dynamic_config::{AcceptTransactionNames, TaggingRule, TransactionMetricsConfig};
 use relay_general::protocol::{
     AsPair, Context, ContextInner, Event, EventType, TraceContext, TransactionSource, User,
 };
 use relay_general::store;
 use relay_general::types::Annotated;
 use relay_metrics::{
-    AggregatorConfig, DurationUnit, Metric, MetricNamespace, MetricUnit, MetricValue,
+    AggregatorConfig, CounterType, DistributionType, DurationUnit, Metric, MetricNamespace,
+    MetricResourceIdentifier, MetricType, MetricValue,
 };
 
 use crate::metrics_extraction::conditional_tagging::run_conditional_tagging;
 use crate::statsd::RelayCounters;
 use crate::utils::SamplingResult;
 
+/// Enumerates the most common transaction-names, with a catch-all for any other names.
+#[derive(Clone, Debug, PartialEq)]
+enum TransactionsMetric {
+    User {
+        value: String,
+        timestamp: UnixTimestamp,
+        tags: TransactionsUserTags,
+    },
+    Duration {
+        unit: DurationUnit,
+        value: DistributionType,
+        timestamp: UnixTimestamp,
+        tags: TransactionsDurationTags,
+    },
+    CountPerRootProject {
+        value: CounterType,
+        timestamp: UnixTimestamp,
+        tags: TransactionsCPRTags,
+    },
+    Breakdowns {
+        name: String,
+        timestamp: UnixTimestamp,
+        value: DistributionType,
+        tags: TransactionsBreakdownTags,
+    },
+    Measurements {
```

Comment:
This represents one measurement, not multiple.

```suggestion
    Measurement {
```